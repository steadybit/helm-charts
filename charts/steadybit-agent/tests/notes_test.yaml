templates:
  - NOTES.txt
tests:
  - it: should warn about missing agent key and cluster name
    set:
      agent:
        auth:
          provider: agent-key
    asserts:
      - matchSnapshot: {}
  - it: should warn about missing agent key
    set:
      global:
        clusterName: test
      agent:
        auth:
          provider: agent-key
    asserts:
      - matchSnapshot: {}
  - it: should warn about missing cluster name
    set:
      agent:
        key: abcdefg
        auth:
          provider: agent-key
    asserts:
      - matchSnapshot: {}
  - it: should inform about deployment
    set:
      global:
        clusterName: test
      agent:
        key: abcdefg
        auth:
          provider: agent-key
    asserts:
      - matchSnapshot: {}
  - it: should warn about deprecated values
    set:
      global:
        clusterName: test
      agent:
        key: abcdefg
        auth:
          provider: agent-key
        extensions:
          autodiscovery:
            matchLabels:
              app: test
            matchLabelsExclude:
              app: testexclude
            namespace: nice-one
    asserts:
      - matchSnapshot: {}
  - it: should inform about missing limits
    set:
      global:
        clusterName: test
      agent:
        key: abcdefg
      resources:
        limits:
          memory: null
          cpu: null
    asserts:
      - matchSnapshot: {}