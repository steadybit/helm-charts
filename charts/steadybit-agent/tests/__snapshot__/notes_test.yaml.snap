should inform about deployment:
  1: |
    raw: |
      It may take a few moments for the agents to fully deploy. You can see what agents are running by listing resources in the NAMESPACE namespace:

          kubectl get all -n NAMESPACE

      You can get the logs for all of the agents with `kubectl logs`:

          kubectl logs -l app.kubernetes.io/name=RELEASE-NAME -n NAMESPACE -c steadybit-agent
should warn about deprecated values:
  1: |
    raw: |
      It may take a few moments for the agents to fully deploy. You can see what agents are running by listing resources in the NAMESPACE namespace:

          kubectl get all -n NAMESPACE

      You can get the logs for all of the agents with `kubectl logs`:

          kubectl logs -l app.kubernetes.io/name=RELEASE-NAME -n NAMESPACE -c steadybit-agent

      #### WARNING: agent.extensions.autodiscovery.matchLabels is deprecated please use agent.extensions.autoregistration.matchLabels instead. ####

      #### WARNING: agent.extensions.autodiscovery.matchLabelsExclude is deprecated please use agent.extensions.autoregistration.matchLabelsExclude instead. ####

      #### WARNING: agent.extensions.autodiscovery.namespace is deprecated please use agent.extensions.autoregistration.namespace instead. ####
should warn about missing agent key:
  1: |
    raw: |
      ##############################################################################
      ####    ERROR: You did not specify your secret agent key.                 ####
      ##############################################################################

      This agent deployment will be incomplete until you set your agent key:

          helm upgrade RELEASE-NAME --reuse-values \
              --set agent.key=$(STEADYBIT_AGENT_KEY) steadybit/steadybit-agent

      - STEADYBIT_AGENT_KEY can be obtained from here: https://platform.steadybit.io/settings/agents/setup
should warn about missing agent key and cluster name:
  1: |
    raw: |2
      ##############################################################################
      ####    ERROR: You did not specify your secret agent key.                 ####
      ####    ERROR: You also did not specify a name for this cluster.          ####
      ##############################################################################

      This agent deployment will be incomplete until you set your agent key and a name for this cluster:

          helm upgrade RELEASE-NAME --reuse-values \
              --set agent.key=$(YOUR_SECRET_AGENT_KEY) \
              --set global.clusterName=$(YOUR_CLUSTER_NAME) steadybit/steadybit-agent

      - STEADYBIT_AGENT_KEY can be obtained from here: https://platform.steadybit.io/settings/agents/setup
      - YOUR_CLUSTER_NAME should be the custom name of your cluster.

      This cluster will be reported as 'unknown' unless you specify a cluster name.
should warn about missing cluster name:
  1: |
    raw: |2
      ##############################################################################
      ####    ERROR: You did not specify a name for this cluster.               ####
      ##############################################################################

      This agent deployment will be incomplete until you set your agent key and a name for this cluster:

          helm upgrade RELEASE-NAME --reuse-values \
              --set global.clusterName=$(YOUR_CLUSTER_NAME) steadybit/steadybit-agent

      - YOUR_CLUSTER_NAME should be the custom name of your cluster.

      This cluster will be reported as 'unknown' unless you specify a cluster name.
